core/src/main/scala/za/co/absa/spline/harvester/postprocessing/extra/ExtraTemplateParser.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spline.harvester.postprocessing.extra
+
+
+import za.co.absa.spline.harvester.postprocessing.extra.ExtraTemplateModel._
+
+import javax.script.ScriptEngineManager
+
+object ExtraTemplateParser {
+
+  private val jsEngine = new ScriptEngineManager().getEngineByMimeType("text/javascript")
```

Comment:
According to the `jsEngine.getFactory.getParameter("THREADING")` return value `null` the engie instance isn't thread-safe (see https://docs.oracle.com/javase/8/docs/api/javax/script/ScriptEngineFactory.html#getParameter-java.lang.String-)
Although in practice, the race-condition should not occur (due to the fact that Spline harvester is invoked from the Spark event thread in a synchronous manner), it nevertheless is not a good practice if a utility class like this makes such assumptions about the global environment and usage specifics.
The thread-safeness should be either guaranteed by the implementation, or otherwise communicated to and managed by the caller of the `parse()` method.